src/main/java/dk/xam/jbang/Main.java

```diff
@@ -263,6 +263,61 @@ public Integer call() throws IOException {
 		}
 	}
 
+	// Similar code in ModularClassPath.java in the method supportsModules file
+	static String getJavaVersion(String command) {
+		ProcessBuilder pb = new ProcessBuilder();
+		String value = "";
+		try {
+			Process p = pb.command(command + "-version").start();
+			p.waitFor();
+			try (BufferedReader bri = new BufferedReader(new InputStreamReader(p.getErrorStream()))) {
+				value = bri.readLine();
+			}
+		} catch (IOException | InterruptedException e) {
+			value = System.getProperty("java.version");
+		}
+		return value;
+	}
+
+	static boolean checkVersions() {
+		String javaVersion = "";
+		String javacVersion = "";
+		try {
+			String x = resolveInJavaHome("java");
+			if (x.contains(" ")) {
+				x = '"' + x + '"';
+			}
+			javaVersion = getJavaVersion(x);
+			Process p = Runtime.getRuntime().exec(resolveInJavaHome("javac -version"));
```

Comment:
this doesn't look right - "-version" is a argument and should be added as such rather than part of the command?

Reply:
I swithced to using `ProcessBuilder` to run the process rather than `Runtime.getRuntime` and passed it a list of arguments.